Routing.md

```diff
@@ -0,0 +1,59 @@
+# Routing Metadata Payload Extension
+
+## Introduction
+Specify a metadata payload for [RSocket](https://rsocket.io) that contains both information about where to route a [`PAYLOAD` frame](https://github.com/rsocket/rsocket/blob/master/Protocol.md#frame-payload) as well as metadata for the frame itself.
+
+## Metadata Payload
+The metadata payload has a MIME Type of `message/x.rsocket.routing`.
+
+### Payload Contents
+```
+     0                   1                   2                   3
+     0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    |            Version            |T|P|S|D|   Flags   |
+    +-------------------------------+-+-+-+-+-----------+-----------+
+    |      Route Token Length       |   Route Token                ...
+    +-------------------------------+-------------------------------+
+    |                                                               |
+    +                                                               +
+    |                                                               |
+    +                           Trace ID                            +
+    |                                                               |
+    +                                                               +
+    |                                                               |
+    +---------------------------------------------------------------+
+    |                                                               |
+    +                           Span ID                             +
+    |                                                               |
+    +---------------------------------------------------------------+
+    |                                                               |
+    +                        Parent Span ID                         +
+    |                                                               |
+    +-+-+-----------------------------------------------------------+
+    |S|D|
+    +-+-+-----------+-----------------------------------------------+
+    |  MIME Length  |   Metadata Encoding MIME Type                ...
+    +---------------+-----------------------------------------------+
+    |  MIME Length  |   Data Encoding MIME Type                    ...
+    +---------------+-----------------------------------------------+
+                          Metadata Payload
+```
+
+* **Version**: (16 bits = max value 65,535) Unsigned 16-bit integer of version number of the routing payload
+* **Flags**: (10 bits)
+  * (**T**)racing: Payload contains distributed tracing tokens
```

Comment:
Am i correct to note that will provide a one time payload for tracing all messages in request/stream or channel scenarios ? Put differently, is this going to trace request/stream and channel entirely vs per-message basis.

Reply:
This payload would be part of the metadata of each `PAYLOAD` frame.  So each `PAYLOAD` _could_ have it's own trace ID within a parent span if desired.  I'm not sure what we want to happen here, so it might be worthwhile to explore and decide if we want to specify something.

Reply:
Moving out into a tracing spec.